File: src/main/java/com/heybys/oddments/fooddelivery/persistence/shop/MenuRepositoryImpl.java
package com.heybys.oddments.fooddelivery.persistence.shop;

import java.util.List;

import org.springframework.stereotype.Repository;

import com.heybys.oddments.base.jpa.BaseRepository;
import com.heybys.oddments.fooddelivery.domain.shop.Menu;
import com.heybys.oddments.fooddelivery.domain.shop.MenuId;
import com.heybys.oddments.fooddelivery.domain.shop.MenuRepository;
import com.heybys.oddments.fooddelivery.domain.shop.ShopId;

@Repository
class MenuRepositoryImpl extends BaseRepository<Menu, MenuId, MenuJpaRepository> implements MenuRepository {

    public MenuRepositoryImpl(MenuJpaRepository repository) {
        super(repository);
    }

    @Override
    public List<Menu> findOpenMenusIn(ShopId shopId) {
        return repository.findByShopIdAndOpenIsTrue(shopId);
    }

    @Override
    public List<Menu> find(ShopId shopId) {
        return repository.findByShopId(shopId);
    }
}

File: src/main/kotlin/com/heybys/oddments/fooddelivery/persistence/shop/MenuRepositoryImpl.kt
package com.heybys.oddments.fooddelivery.persistence.shop

import com.heybys.oddments.base.jpa.BaseRepository
import com.heybys.oddments.fooddelivery.domain.shop.Menu
import com.heybys.oddments.fooddelivery.domain.shop.MenuId
import com.heybys.oddments.fooddelivery.domain.shop.MenuRepository
import com.heybys.oddments.fooddelivery.domain.shop.ShopId
import org.springframework.stereotype.Repository

@Repository
internal class MenuRepositoryImpl(repository: MenuJpaRepository) :
    BaseRepository<Menu, MenuId, MenuJpaRepository>(repository), MenuRepository {

    override fun findOpenMenusIn(shopId: ShopId): List<Menu> {
        return repository.findByShopIdAndIsOpenIsTrue(shopId)
    }

    override fun find(shopId: ShopId): List<Menu> {
        return repository.findByShopId(shopId)
    }
}
